pkg/sql/sem/builtins/builtins.go

```diff
@@ -3064,6 +3064,29 @@ may increase either contention or retry errors, or both.`,
 		},
 	),
 
+	"getdatabaseencoding": makeBuiltin(
+		tree.FunctionProperties{Category: categorySystemInfo},
+		tree.Overload{
+			Types:      tree.ArgTypes{},
+			ReturnType: tree.FixedReturnType(types.String),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				// encoding name has been computed bypassing encoding_id
+				// extracted from pg_database to pg_encoding_to_char function.
+				row, err := ctx.InternalExecutor.QueryRow(
+					ctx.Ctx(), "getdatabaseencoding",
+					ctx.Txn,
+					"SELECT pg_encoding_to_char(encoding) "+
```

Comment:
Can this not just be `SELECT pg_encoding_to_char(encoding) from pg_database...`? Why do you need a subquery?

Reply:
```select pg_encoding_to_char(encoding) from pg_database where datname = current_database()```
sorry my bad we can work with above commmnd, without creating subquery.

Reply:
```select pg_encoding_to_char(encoding) from pg_database where datname = $1```
sorry my bad we can work with above commmnd, without creating subquery.

I am making these changes too.